### DISCUSSION TITLE
some questions about load plugins in lazy.nvim

### DISCUSSION THREAD

**Discussion by shellRaining** [askfiy/nvim#24] [Q&A]
hey, pleasantly surprised to see that your neovim configuration has been refactored. This refactoring makes the code clearer, but I still have some issues

1. How do you deal with plugins such as `catppuccin/nvim`
[https://github.com/askfiy/nvim/blob/master/lua/utils/aid/lazy.lua#L49](https://github.com/askfiy/nvim/blob/master/lua/utils/aid/lazy.lua#L49), the `require_file_name` will be split to nvim, and that means in theme dir there will have a file named nvim.lua if i use this plugin

2. why use load function to execute require(...).before()
[https://github.com/askfiy/nvim/blob/master/lua/utils/aid/lazy.lua#L57](https://github.com/askfiy/nvim/blob/master/lua/utils/aid/lazy.lua#L57), this load function seems return a nil value, which means `plugin_opts.config` is nil, can i refactor this code from 

```lua
plugin_opts.config = load(([[
    require("utils.api").require_all_package(require("%s"))
    require("%s").load()
    require("%s").after()
]]):format(require_file_path, require_file_path, require_file_path))
```

to
```lua
require("utils.api").require_all_package(module)
module.load()
module.after()
```

**Comment by askfiy** [comment]
1. Lazy.nvim has a name option, which is the same as option as in Packer. You just need to specify name = true for catppuccin/nvim.

2. In desperation, init must be a function. Inside a local function, I can't get a module... This is Lua's magical place...

**Comment by askfiy** [comment]
It seems that `lazy.nvim` is not the same as `packer.nvim`, I can use `module` directly in the `init` and `config` functions of `lazy.nvim`, which is much more convenient, without `load( fun_string)`.
